pkg/network/setup/network.go

```diff
@@ -121,16 +122,38 @@ func (n *VMNetworkConfigurator) SetupPodNetworkPhase2(domain *api.Domain, networ
 }
 
 func preConfigStateRun(nics []podNIC) ([]podNIC, error) {
-	return discoverPodInterfaces(nics)
+	nics, err := discoverPodInterfaces(nics)
```

Comment:
Can we store the `podInterfaceName` return by `discoverPodInterfaces` in the ConfigState, so we won't need to invoke it each time. Then if the ConfigState.PodIntefaceName` is not empty and does'nt pass the filter, we can avoid stepping in the virt-launcher ns?